### PR TITLE
Feature/add documentation

### PR DESCRIPTION
**Pull request by nburkley**
Adding documentation and comments to the codebase

### CODE REVIEW

**Comment by nburkley** [comment]
ping @edenspiekermann/backend

File: lib/espi_dni/analytics_manager.ex [hunk 1]
```diff
@@ -9,6 +14,17 @@ defmodule EspiDni.AnalyticsManager do
     ArticleViewSourceNotifier
   }
 
+  @doc """
+  Makes a call to Google anlaytics using GoogleRealtimeClient to get the recent
```
Comment (zfhui) [inline lib/espi_dni/analytics_manager.ex]: Capitalization: **G**oogle **A**nalytics, also in L21.

File: lib/espi_dni/analytics_manager.ex [hunk 2]
```diff
@@ -9,6 +14,17 @@ defmodule EspiDni.AnalyticsManager do
...
+  pageviews for all articles belonging to a team.
+
+  The viewcounts retrieved from google analytics are the processed, and saved
+  if the correspondning article is found.
```
Comment (zfhui) [inline lib/espi_dni/analytics_manager.ex]: This sentence sound a bit funny. 

Also: typo: `corresponding`.

File: lib/espi_dni/article_slack_messenger.ex [hunk 3]
```diff
@@ -1,26 +1,38 @@
...
+
   alias EspiDni.{
     SlackWeb,
     Repo,
     NotificationMessage
   }
   import EspiDni.Gettext
 
+  @doc """
+  Sends a view spike message to an articles user
```
Comment (zfhui) [inline lib/espi_dni/article_slack_messenger.ex]: `article's`, also L25

File: lib/espi_dni/article_view_count_notifier.ex [hunk 4]
```diff
@@ -20,18 +29,22 @@ defmodule EspiDni.ArticleViewCountNotifier do
...
       select: sum(view_count.count),
-      where: view_count.article_id == ^article.id,
       where: view_count.inserted_at > ago(2, "hour"),
       group_by: fragment("round(extract('epoch' from inserted_at) / 1800)"),
       order_by: fragment("round(extract('epoch' from inserted_at) / 1800) desc"),
       limit: 2
     )
   end
 
+  # Decides if the different between the current and previous view counts
```
Comment (zfhui) [inline lib/espi_dni/article_view_count_notifier.ex]: `difference`?

File: lib/espi_dni/google_analytics_client.ex [hunk 5]
```diff
@@ -9,6 +13,11 @@ defmodule EspiDni.GoogleAnalyticsClient do
   @client_id Application.get_env(:ueberauth, Ueberauth.Strategy.Google.OAuth)[:client_id]
   @client_secret Application.get_env(:ueberauth, Ueberauth.Strategy.Google.OAuth)[:client_secret]
 
+  @doc"""
```
Comment (zfhui) [inline lib/espi_dni/google_analytics_client.ex]: Missing space between `@doc`and `"""`, also in L34

File: lib/espi_dni/google_analytics_client.ex [hunk 6]
```diff
@@ -55,6 +71,13 @@ defmodule EspiDni.GoogleAnalyticsClient do
     %{"access_token" => team.google_token}
   end
 
+  # the params required to refresh a google oauth2 token
+  # refresh_token: A token required to request new oaut2 tokens
+  # client_id: The google client_id of the appilcation making the request
+  # (this app)
+  # client_secret: The google client_secret of the appilcation making the
+  # request (this app)
+  # grant_type: the type of authorisation request being made
```
Comment (zfhui) [inline lib/espi_dni/google_analytics_client.ex]: Would indentation from `:` on make this more readable?

File: lib/espi_dni/slack_rtm.ex [hunk 7]
```diff
@@ -1,25 +1,43 @@
...
+  """
+
   use Slack
   require Logger
 
   @token Application.get_env(:espi_dni, __MODULE__)[:slack_token]
 
   def start_link, do: start_link(@token)
 
+  @doc"""
```
Comment (zfhui) [inline lib/espi_dni/slack_rtm.ex]: Missing space between `@doc` and `"""`, also at several other places in this file.

File: lib/espi_dni/slack_rtm.ex [hunk 8]
```diff
@@ -1,25 +1,43 @@
...
+  Called when a message is received
+  """
   def handle_message(message = %{type: "message"}, slack) do
     if message.user != slack.me.id do
       Logger.debug "Message received: #{slack.me.name}"
     end
   end
 
+  @doc"""
+  Called when any other (not an message) slack activity is received
```
Comment (zfhui) [inline lib/espi_dni/slack_rtm.ex]: `Slack` was capitalized everywhere else.

File: lib/espi_dni/slack_web.ex [hunk 9]
```diff
@@ -1,10 +1,19 @@
...
+  Uses https://github.com/BlakeWilliams/Elixir-Slack
+  """
+
   use Slack
 
   import Ecto.Query, only: [from: 1, from: 2]
   alias EspiDni.Repo
   alias EspiDni.Team
 
+  @doc"""
```
Comment (zfhui) [inline lib/espi_dni/slack_web.ex]: Missing space.

File: lib/espi_dni/unique_queue.ex [hunk 10]
```diff
@@ -23,7 +36,8 @@ defmodule EspiDni.UniqueQueue do
   end
 
   defp worker_name(module_name) do
-    Atom.to_string(module_name)
+    module_name
```
Comment (zfhui) [inline lib/espi_dni/unique_queue.ex:39]: 😍

File: web/models/google_web_property.ex [hunk 11]
```diff
@@ -1,4 +1,9 @@
 defmodule EspiDni.GoogleWebProperty do
+
+  @moduledoc """
+  A module representing a google analytics web property account
```
Comment (zfhui) [inline web/models/google_web_property.ex]: Google Analytics.

File: web/models/slack_auth_handler.ex [hunk 12]
```diff
@@ -1,13 +1,20 @@
 defmodule EspiDni.SlackAuthHandler do
 
   @moduledoc """
-  Create or retreive the team and user from an auth request
+  Create or retrieve the team and user from a slack oauth response
```
Comment (zfhui) [inline web/models/slack_auth_handler.ex:4]: Slack

File: web/models/team.ex [hunk 13]
```diff
@@ -53,6 +63,9 @@ defmodule EspiDni.Team do
     )
   end
 
+  @doc """
+  Returns an atom representing how far along the setup process the team is
```
Comment (zfhui) [inline web/models/team.ex]: `the setup process *of* the team is`?

File: web/plugs/require_slack_token.ex [hunk 14]
```diff
@@ -1,5 +1,11 @@
 defmodule EspiDni.Plugs.RequireSlackToken do
 
+  @moduledoc """
+  Checks if the Slack token is present (a secret token that should be present
+  on all requests coming from slack verifying it is a real requeat)
```
Comment (zfhui) [inline web/plugs/require_slack_token.ex]: Slack

File: web/plugs/set_slack_user.ex [hunk 15]
```diff
@@ -1,5 +1,12 @@
 defmodule EspiDni.Plugs.SetSlackUser do
 
+  @moduledoc """
+  Finds the team from the slack params
```
Comment (zfhui) [inline web/plugs/set_slack_user.ex]: Slack.

**Review by zfhui — APPROVED** [review]
I didn't find anything major, just a few inconsistencies for capitalizing words. 

Looks good. Will you generate docs based on those comments?

**Review by martzoukos — APPROVED** [review]
👍

**Comment by nburkley** [comment]
Thanks @zfhui, & @martzoukos :)

Documentation is a first class citizen in Elixir - you don't need to generate documentation from it, it's becomes part of the code itself. https://hexdocs.pm/elixir/writing-documentation.html

**Pull request merged**